src/BlockService/CheckoutBlockService.php

```diff
@@ -68,17 +75,51 @@ public function availableGateways()
         if ($filters) {
             WC()->customer->set_billing_country($billingCountry);
             $availableGateways = WC()->payment_gateways()->get_available_payment_gateways();
-            foreach ($availableGateways as $key => $gateway){
-                if(strpos($key, 'mollie_wc_gateway_') === false){
-                    unset($availableGateways[$key]);
-                }
-            }
+            $availableGateways = $this->removeNonMollieGateway($availableGateways);
+            $availableGateways = $this->maybeRemoveVoucher($availableGateways);
```

Comment:
I'm probably missing something, but instead of grabbing *all* payment gateways from globals and _then_ filtering out everything BUT our own (and _maybe_ the voucher gateway), would it not be simpler (as in: less complex and easier to test) to add these two gateways as constructor arguments and use them accordingly here - especially since we're really only interested in their IDs. So you could technically go even further and just pass those IDs as `string` into the constructor :)

Reply:
Hmm, not sure how would that work... The thing here is we are looking for the available gateways (that can be ours or not) and the list changes depending on the cart contents and the customer details (country selection changing in the block). So, `get_available_payment_gateways()` does not retrieve all, but all that can be used (after API call ) I could do it myself instead of using the global and go by all Mollie gateways and ask if they are available, instead of retrieving the available and removing... do you think that would work better?